paddle/fluid/inference/tests/api/analyzer_bert_tester.cc

```diff
@@ -228,5 +229,44 @@ TEST(Analyzer_bert, compare_determine) {
   CompareDeterministic(reinterpret_cast<const PaddlePredictor::Config *>(&cfg),
                        inputs);
 }
+
+TEST(Analyzer_bert, transfer_scope_cache) {
+  AnalysisConfig config;
+  SetConfig(&config);
+
+  std::vector<PaddleTensor> input, output;
+  auto predictor = CreatePaddlePredictor<AnalysisConfig>(config);
+
+  int threads_num = 10;
+  std::vector<std::thread> threads;
+  std::unordered_set<std::unordered_set<paddle::framework::Scope *> *>
+      global_transfer_scope_cache;
+  std::unordered_set<std::unordered_map<size_t, paddle::framework::Scope *> *>
+      global_transfer_data_cache;
+
+  std::ifstream fin(FLAGS_infer_data);
+  std::string line;
+
+  for (int i = 0; i < threads_num; i++) {
+    threads.emplace_back([&, i]() {
+      std::getline(fin, line);
+      ParseLine(line, &input);
+      predictor->Run(input, &output, FLAGS_batch_size);
+      global_transfer_scope_cache.insert(
+          &paddle::framework::global_transfer_scope_cache());
+      global_transfer_data_cache.insert(
+          &paddle::framework::global_transfer_data_cache());
+    });
+    threads[0].join();
+    threads.clear();
+    std::vector<PaddleTensor>().swap(input);
+  }
+  // Since paddle::framework::global_transfer_scope_cache() and
+  // paddle::framework::global_transfer_data_cache() are thread_local,
+  // their pointer should be different among different thread id.
+  PADDLE_ENFORCE(global_transfer_scope_cache.size(), threads_num);
+  PADDLE_ENFORCE(global_transfer_data_cache.size(), threads_num);
```

Comment:
The size should be 1 in #18428 specific mkldnn tid setting

Reply:
is the cpu_num set -1 for MKLDNN tested?

Reply:
Since #18453 refine api and const variable name to make it sense. Then 
- #18428 will follow the new api `platform:: set_cur_mkldnn_session_id(-1)` (not cpu_num) for MKLDNN tested. See:
https://github.com/PaddlePaddle/Paddle/compare/develop...luotao1:LeoZhao-Intel-mt_memoryleak_test?expand=1#diff-db5799d22e4420bb8cdacdad8d5e2b67R255
```
if (enable_mkldnn_cache_clear_mode) ``; 
...
if (enable_mkldnn_cache_clear_mode) {
    PADDLE_ENFORCE(global_transfer_scope_cache.size(), 1);
  } else {
    PADDLE_ENFORCE(global_transfer_scope_cache.size(), threads_num);
  }
```
- After #18428 merged, we will enhance the `config.EnableMKLDNN()` interface, and remove 
`platform::set_cur_mkldnn_session_id(-1)` in the inference codes.